google/cloud/bigtable/table.cc

```diff
@@ -171,6 +172,7 @@ std::vector<FailedMutation> Table::BulkApply(BulkMutation mut) {
     if (!status.ok() && !retry_policy->OnFailure(status)) {
```

Comment:
My proposal was:

```cc
while (mutator.HasPendingMutations()) { 
....  
  status = mutator.MakeOneRequest(*client_, client_context);
  if (status.ok()) continue;
  if (!retry_policy->OnFailure(status)) break;
...
}
```

I think your approach saves the initial `HasPendingMutations()` probably worth it.